docs/builtin-objects.md

```diff
@@ -0,0 +1,56 @@
+# environment template built-in objects
```

Comment:
```suggestion
# helmfile template built-in objects
```

We have ambiguity on how to call it- helmfile, helmfile.yaml, helmfile config, and helmfile state file.

Since v1, it isn't going to be a yaml file so we shouldn't call it "helmfile.yaml".

helmfile "config" and "state" don't make much sense to me now as it's not the only config/state file if we consider it carefully- even a release values file might be conceptually a "config" or "state" file dedicated to a set of release values.

That said, "helmfile" looks to be the only viable option, although it doesn't sound perfect... 😢

Reply:
I've used to call it "release set" in https://github.com/mumoshu/terraform-provider-helmfile.

If you like it, I think we can fully migrate to that starting v1. It might be documentation-only migration as I presume we don't have any occurrences of "state file" or "helmfile config file" in code and logs.

```suggestion
# releaseset template built-in objects
```

Reply:
Well, we already call it "helmfile" for the `helmfiles` field in helmfile.yaml. Each helmfiles entry is called "sub-helmfile".

Shall we use "sub-releaseset" instead of "sub-helmfile"? 🤔 I'm not confident about what's best.